core/src/main/scala/prometheus4cats/Summary.scala

```diff
@@ -0,0 +1,287 @@
+      *   if `string` is not valid
+      */
+    def unsafeFrom(value: Double): Quantile =
+      from(value).fold(msg => throw new IllegalArgumentException(msg), identity)
+
+    implicit val catsInstances: Hash[Quantile] with Order[Quantile] with Show[Quantile] = new Hash[Quantile]
+      with Order[Quantile]
+      with Show[Quantile] {
+      override def hash(x: Quantile): Int = Hash[Double].hash(x.value)
+
+      override def compare(x: Quantile, y: Quantile): Int = Order[Double].compare(x.value, y.value)
+
+      override def show(t: Quantile): String = Show[Double].show(t.value)
+
+      override def eqv(x: Quantile, y: Quantile): Boolean = Eq[Double].eqv(x.value, y.value)
+    }
+  }
+
+  final case class QuantileDefinition private (value: Quantile, error: Double) {
+    override def toString: String = s"""Summary.QuantileDefinition(value: "${value.value}", error: "$error")"""
+  }
+
+  object QuantileDefinition extends internal.SummaryQuantileDefinitionFromDoubleLiterals {
+
+    /** Parse a [[QuantileDefinition]] from the given string
+      *
+      * @param value
+      *   quantile value
+      * @param error
+      *   error rate of the given quantile
+      * @return
+      *   a parsed [[QuantileDefinition]] or failure message, represented by an [[scala.Either]]
+      */
+    def from(value: Double, error: Double): Either[String, QuantileDefinition] = Quantile.from(value).flatMap { v =>
+      Either.cond(
+        error >= 0.0 && error <= 1.0,
+        new QuantileDefinition(v, error),
+        s"Quantile error rate $error must be between 0.0 and 1.0"
+      )
+    }
+
+    /** Unsafely parse a [[QuantileDefinition]] from the given values
+      *
+      * @param value
+      *   value from which to parse a quantile value
+      * @param error
+      *   error rate of the given quantile
+      * @return
+      *   a parsed [[QuantileDefinition]]
+      * @throws java.lang.IllegalArgumentException
+      *   if `string` is not valid
+      */
+    def unsafeFrom(value: Double, error: Double): QuantileDefinition =
+      from(value, error).fold(msg => throw new IllegalArgumentException(msg), identity)
+
+    implicit val catsInstances: Hash[QuantileDefinition] with Order[QuantileDefinition] with Show[QuantileDefinition] =
+      new Hash[QuantileDefinition] with Order[QuantileDefinition] with Show[QuantileDefinition] {
+        override def hash(x: QuantileDefinition): Int = Hash[(Quantile, Double)].hash(x.value -> x.error)
+
+        override def compare(x: QuantileDefinition, y: QuantileDefinition): Int =
+          Order[(Quantile, Double)].compare(x.value -> x.error, y.value -> y.error)
+
+        override def show(t: QuantileDefinition): String = t.toString
+
+        override def eqv(x: QuantileDefinition, y: QuantileDefinition): Boolean =
+          Eq[(Quantile, Double)].eqv(x.value -> x.error, y.value -> y.error)
+      }
+  }
+
+  case class Value[A](count: A, sum: A, quantiles: Map[Quantile, A] = Map.empty) {
```

Comment:
I'm not 100% convinded in the need for the refined quantile type here, as this will likely be received at runtime

Reply:
Answering my own question: changed this to a `Double`